netty/src/test/java/io/grpc/netty/NettyStreamTestBase.java

```diff
@@ -129,11 +133,12 @@ public void inboundMessageShouldCallListener() throws Exception {
       ((NettyClientStream) stream).transportState()
           .transportDataReceived(messageFrame(MESSAGE), false);
     }
-    ArgumentCaptor<InputStream> captor = ArgumentCaptor.forClass(InputStream.class);
-    verify(listener()).messageRead(captor.capture());
+
+    InputStream message = listenerMessageQueue.poll();
```

Comment:
This being populated by the extending class is really subtle. I think we've instead made an abstract method here (say, `listenerMessageQueue()`) that is required to be implemented by extending classes.